dapp/src/components/shared/NumberFormatInput/index.tsx

```diff
@@ -29,17 +35,35 @@ const NumberFormatInput = React.forwardRef<
 >((props, ref) => {
   const { field: css } = useMultiStyleConfig("Input", props)
 
-  const { decimalScale, isDisabled, isInvalid, ...restProps } = props
+  const {
+    decimalScale,
+    isDisabled,
+    isInvalid,
+    maxIntegerLength,
+    maxDecimalLength,
+    ...restProps
+  } = props
+
+  const handleLengthValidation = (values: NumberFormatValues) => {
+    const { value } = values
+    if (!value || !maxIntegerLength || !maxDecimalLength) return true
+
+    const [integer, decimal] = value.split(".")
+    const isValidIntegerLength = integer && integer.length <= maxIntegerLength
+    const isValidDecimalLength = decimal && decimal.length <= maxDecimalLength
+
+    return decimal ? isValidDecimalLength : isValidIntegerLength
+  }
```

Comment:
When we get rid of `maxDecimalLength` we can simplify it a bit. We want to check the integer part from the value. So we can use the `parseInt` method. What do you think about it?

I noticed two more things:

1. The `!maxIntegerLength` return `true` when `maxIntegerLength=0`. I think in such a case we would want to move on instead of exiting the function.
2. The user can enter the value `.` to input. That's why I would check if `floatValue` is defined to correctly convert the `value` to integer.

```suggestion
  const handleLengthValidation = (values: NumberFormatValues) => {
    const { value, floatValue } = values

    if (floatValue === undefined || maxIntegerLength === undefined) return true

    const convertedValue = parseInt(value, 10)
    const valueLength = convertedValue.toString().length

    return valueLength <= maxIntegerLength
  }
```

Reply:
Ref commit: 880bbd9